sdgx/synthesizer.py

```diff
@@ -334,9 +345,16 @@ def sample(
         logger.info("Sampling...")
         metadata = metadata or self.metadata
         self.metadata = metadata  # Ensure update metadata
-        if metadata:
-            for d in self.data_processors:
-                d.fit(metadata)
+
+        # data_processors do not need to be fit again in the sampling stage
+        # if metadata:
+        #     for d in self.data_processors:
+        #         d.fit(metadata)
+
```

Comment:
OK we can just remove them? Seems only need keep first line
```suggestion
        # data_processors do not need to be fit again in the sampling stage

```